### PR TITLE
Implement E2E tests

### LINKED ISSUE #68: implement E2E tests
### Description

Blocked by https://github.com/Rpc-h/RPCh/issues/69

Using the testing environment, implement an E2E test that tests everything by using the adaptor and waiting for the response.

1. create a DEMO app which uses the ethers adaptor
2. write tests for the DEMO app

### Relevant issues

- [x] create DEMO app
- [x] implement E2E tests
- [ ] update CI to include E2E tests

### Definition of DONE

Once E2E tests are implemented and they are part of our CI.

### PR DESCRIPTION
**Pull request by ItsLuized**
Closes #68

### CODE REVIEW

File: packages/ethers/src/index.spec.ts [hunk 1]
```diff
@@ -10,6 +10,14 @@ const ENTRY_NODE_API_TOKEN = "12345";
 const ENTRY_NODE_PEER_ID = fixtures.PEER_ID_A;
 const EXIT_NODE_PEER_ID = fixtures.PEER_ID_B;
 
+jest.mock("rpch-common", () => ({
+  ...jest.requireActual("rpch-common"),
+  hoprd: {
+    createMessageListener: jest.fn(),
+    sendMessage: jest.fn(),
```
Comment (nionis) [inline packages/ethers/src/index.spec.ts:17]: I think `sendMessage` is already mocked down below, maybe remove it from here?

File: packages/sdk/src/index.ts [hunk 2]
```diff
@@ -137,31 +139,33 @@ export default class SDK {
     this.interval = setInterval(() => {
       this.segmentCache.removeExpired(this.timeout);
       this.requestCache.removeExpired(this.timeout);
-    }, 1000);
+    }, 1e4);
```
Comment (nionis) [inline packages/sdk/src/index.ts:142]: why the change here? `1e4 = 10000`

**Review by nionis — CHANGES_REQUESTED** [review]
- I would like to use the `devkit` folder as a general folder for all things developer + E2E testing, so I would like us to do the following:
  1. `devkit/sandbox` should contain our docker-compose
  2. `devkit/e2e` should contain our E2E test project
- some other minor feedback

File: packages/sdk/src/index.ts (hunk 2 above)
Comment (ItsLuized) [reply packages/sdk/src/index.ts:142]: I didn't notice the mistake. Fixed

File: packages/sdk/src/index.spec.ts [hunk 3]
```diff
@@ -12,6 +12,14 @@ const EXIT_NODE_PEER_ID = fixtures.PEER_ID_B;
 
 type MockOps = HoprSdkTempOps & { timeout: number };
 
+jest.mock("rpch-common", () => ({
+  ...jest.requireActual("rpch-common"),
+  hoprd: {
+    createMessageListener: jest.fn(),
```
Comment (nionis) [inline packages/sdk/src/index.spec.ts:18]: should also return a mocked function which is used to stopping the listener

File: devkit/e2e/package.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  "version": "0.0.0",
+  "main": "./build/index.js",
+  "types": "./build/index.d.ts",
+  "license": "MIT",
+  "scripts": {
+    "build": "exit 0",
+    "dev": "yarn build --watch",
+    "format": "prettier --check \"src/**/*.{ts,md}\"",
+    "lint": "TIMING=1 eslint \"src/**/*.ts*\"",
+    "test": "exit 0",
```
Comment (nionis) [inline devkit/e2e/package.json:12]: can we make the test function here run the sh script & all the tests?

File: package.json [hunk 5]
```diff
@@ -5,7 +5,8 @@
   "workspaces": [
     "configs/*",
     "packages/*",
-    "apps/*"
+    "apps/*",
+    "devkit/e2e"
```
Comment (nionis) [inline package.json:9]: is there an issue when using `devkit/*`?

**Review by nionis — CHANGES_REQUESTED** [review]
- some feedback
- please also resolve conflict with `main` branch

File: package.json (hunk 5 above)
Comment (ItsLuized) [reply package.json:9]: Doesn't present any issue and keeps the styles of the other workspaces.
Fixed ✅

File: devkit/e2e/package.json (hunk 4 above)
Comment (ItsLuized) [reply devkit/e2e/package.json:12]: Good idea! Done ✅

**Review by nionis — APPROVED** [review]
